src/irr/asset/CImageLoaderOpenEXR.cpp

```diff
@@ -0,0 +1,454 @@
+			else if (format == EF_R32G32B32A32_SFLOAT || format == EF_R32G32B32_SFLOAT)
+				pixelType = PixelType::FLOAT;
+			else if (format == EF_R32G32B32A32_UINT || format == EF_R32G32B32_UINT)
+				pixelType = PixelType::UINT;
+
+			for (uint8_t rgbaChannelIndex = 0; rgbaChannelIndex < (doesItSupportAlphaChannel ? 4 : 3); ++rgbaChannelIndex)
+				frameBuffer.insert
+				(
+					reinterpret_cast<const char*>(rgbaSignatureAsText[rgbaChannelIndex]),                       // name
+					Slice(pixelType,                                                                            // type
+					(char*)(&(pixelRgbaMapArray[rgbaChannelIndex])[0][0] - dw.min.x - dw.min.y * width),        // base
+					sizeof((pixelRgbaMapArray[rgbaChannelIndex])[0][0]) * 1,                                    // xStride
+					sizeof((pixelRgbaMapArray[rgbaChannelIndex])[0][0]) * width,                                // yStride
+					1, 1,                                                                                       // x/y sampling
+					0                                                                                           // fillValue
+				));	
+
+			file.setFrameBuffer(frameBuffer);
+			file.readPixels(dw.min.y, dw.max.y);
+		}
+
+		void specifyIrrlichtEndFormat(E_FORMAT& format, const InputFile& file, bool& doesItSupportAlphaChannel)
+		{
+			const IMF::Channel* RChannel = file.header().channels().findChannel("R");
+			const IMF::Channel* GChannel = file.header().channels().findChannel("G");
+			const IMF::Channel* BChannel = file.header().channels().findChannel("B");
+			const IMF::Channel* AChannel = file.header().channels().findChannel("A");
+
+			if (!RChannel || !GChannel || !BChannel)
+				assert(0);
+
+			if (AChannel)
+				doesItSupportAlphaChannel = true;
+			else
+				doesItSupportAlphaChannel = false;
+
+			auto doesRGBFormatHaveTheSameFormatLikePassedToIt = [&](const PixelType ImfTypeToCompare)
+			{
+				return (RChannel->type == ImfTypeToCompare && GChannel->type == ImfTypeToCompare && BChannel->type == ImfTypeToCompare);
+			};
+
+			auto doesAChannelExistAndIsTheSameFormatLikePassedToIt = [&](const PixelType ImfTypeToCompare)
+			{
+				return (AChannel && AChannel->type == ImfTypeToCompare);
+			};
+
+			if (doesRGBFormatHaveTheSameFormatLikePassedToIt(PixelType::HALF))
+			{
+				if (doesAChannelExistAndIsTheSameFormatLikePassedToIt(PixelType::HALF))
+					format = EF_R16G16B16A16_SFLOAT;
+				else
+					format = EF_R16G16B16_SFLOAT;
+			}
+			else if (doesRGBFormatHaveTheSameFormatLikePassedToIt(PixelType::FLOAT))
+			{
+				if (doesAChannelExistAndIsTheSameFormatLikePassedToIt(PixelType::FLOAT))
+					format = EF_R32G32B32A32_SFLOAT;
+				else
+					format = EF_R32G32B32_SFLOAT;
+			}
+			else if (doesRGBFormatHaveTheSameFormatLikePassedToIt(PixelType::UINT))
+			{
+				if (doesAChannelExistAndIsTheSameFormatLikePassedToIt(PixelType::UINT))
+					format = EF_R32G32B32A32_UINT;
+				else
+					format = EF_R32G32B32_UINT;
+			}
```

Comment:
the alpha checks will cause a texture with a present but different alpha format to RGB to register as a non-alpha texture (should fail instead)

Reply:
> Did you forget to add and commit `cmake/FindZLIB.` (the override for findpackage zlib)?

no need for that, I fixed CMake and now it works properly :D